### ISSUE TITLE
Core/Spell Victory Rush not scripted in core.

### ISSUE THREAD

**Issue by HannibalRoG** [TrinityCore/TrinityCore#18578]
**Description:**  There is no script in core for Victory Rush.

**Current behaviour:** (Tell us what happens.)  
It is activated by killing an NPC which brings experience and you can use it to do damage while it lasts victorious and does not cure you.  

**Expected behaviour:** (Tell us what should happen instead.)  
You must disappear the victorious buff once you use Victory Rush and you must heal a% of life.

**Steps to reproduce the problem:**

1. Create a Warrior class.  
2. Levelup to 10.  
3. Kill a npc that get xp.  
4. Use Victory Rush.

**Branch(es):** 3.3.5 / master (Tell us which branch(es) this issue affects.)
**master**

**TC rev. hash/commit:**  25bdf4b5291b90f436019bed416aa95864716a55

**TDB version:**  703.00 + updates

**Operating system:**

**Comment by Edreisantafe** [comment]
This fix the problem blizzlike.   

`WARRIOR_SPELL_VICTORY_RUSH_DAMAGE = 34428,  
WARRIOR_SPELL_VICTORY_RUSH_HEAL = 118779,  
WARRIOR_SPELL_VICTORIOUS_STATE = 32216,  

class spell_warr_victory_rush_SpellScript : public SpellScript
    {
        PrepareSpellScript(spell_warr_victory_rush_SpellScript);

        bool Validate(SpellInfo const* /*SpellEntry*/)
        {
            if (!sSpellMgr->GetSpellInfo(WARRIOR_SPELL_VICTORY_RUSH_DAMAGE))
                return false;
            return true;
        }
        void HandleOnHit()
        {
            if (Player* _player = GetCaster()->ToPlayer())
            {
                if (Unit* target = GetHitUnit())
                {
                    if (_player->HasAura(WARRIOR_SPELL_VICTORIOUS_STATE))
                    {
                        _player->CastSpell(_player, WARRIOR_SPELL_VICTORY_RUSH_HEAL, true);
                        _player->RemoveAura(WARRIOR_SPELL_VICTORIOUS_STATE);
                    }
                }
            }
        }

        void Register()
        {
            AfterHit += SpellHitFn(spell_warr_victory_rush_SpellScript::HandleOnHit);
        }
    };

new spell_warr_victory_rush();    

DELETE FROM `spell_script_names` WHERE spell_id = '34428';
  
INSERT INTO `spell_script_names` VALUES (34428, 'spell_warr_victory_rush');  `

**Comment by zamalaev** [comment]
```
            case SPELLFAMILY_WARRIOR:
            {
                switch (m_spellInfo->Id)
                {
                    case 34428: // Victory Rush
                    {
                        if (m_caster->ToPlayer()->GetSpecializationId(m_caster->ToPlayer()->GetActiveSpec()) == SPEC_WARRIOR_ARMS)
                            damage += CalculatePct(m_caster->GetTotalAttackPowerValue(BASE_ATTACK), 67.2f);
                        else
                            damage += CalculatePct(m_caster->GetTotalAttackPowerValue(BASE_ATTACK), 56.0f);

                        break;
                    }
```

```
        case 34428: // Victory Rush
            break;
        default:
            if (spellProto->HasEffect(SPELL_EFFECT_HEAL_PCT))
                return false;
            break;
```

**Comment by zamalaev** [comment]
```
  case 118779: // Glyph of Victory Rush
            if (AuraEffect * glyph = m_caster->GetAuraEffect(58382, EFFECT_0))
                AddPct(damage, glyph->GetAmount());
            break;
```
 ```
              // Victorious
                case 32216:
                {
                    RemoveAura(dummySpell->Id);
                    return false;
                }
```

**Comment by Ovahlord** [comment]
Hack detected. Everything can be calculated within the spellscript.

**Comment by Edreisantafe** [comment]
As far as I know there are no glyphs in Legion, it does not make sense then to program a nonexistent spell.

**Comment by Aokromes** [comment]
there are cosmetic glyphs.

**Comment by Edreisantafe** [comment]
@Aokromes And they need to be scripted in the .cpp?

**Comment by HannibalRoG** [comment]
This was fixed in e5e8537c3ef6cfc5d1bcb9c076a7c75f7e66a619

**Comment by Aokromes** [comment]
you known, you can close your tickets :)
